tests/admin_custom/tests.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import unicode_literals
+
+from django.contrib import admin
+from django.contrib.admin import sites
+from django.test import SimpleTestCase, override_settings
+
+
+INSTALLED_APPS = [
+    'admin_custom.apps.MyCustomAdminConfig',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+]
+
+
+@override_settings(INSTALLED_APPS=INSTALLED_APPS)
+class CustomAdminSiteTests(SimpleTestCase):
+    """
+    Because django.contrib.admin.site can be already instantiated by another
+    test (ie admin_changelist) when running the full test suite,
+    temporary reset with a new one.
+    """
+
+    def setUp(self):
+        self._old_site = admin.site
+        admin.site = sites.site = sites.DefaultAdminSite()
+
+    def tearDown(self):
+        admin.site = sites.site = self._old_site
+
+    def test_use_custom_admin_site(self):
+        self.assertEqual(hasattr(admin.site, '_custom'), True)
+        self.assertEqual(admin.site._custom, True)
+        self.assertEqual(admin.site.__class__.__name__, 'CustomAdminSite')
```

Comment:
Both the `_custom` and `__class__.__name__` assertions are testing the same thing. I'd use just the last one. (N.B. for the first one it's also unnecessary to do both `hasattr` and then compare the attribute to the expected value, you can just do the compare and the test will fail with `AttributeError`)